main.go

```diff
@@ -1,20 +1,82 @@
 package main
 
 import (
-	"github.com/dmacvicar/terraform-provider-libvirt/libvirt"
-	"github.com/hashicorp/terraform/plugin"
+	"flag"
+	"fmt"
+	"log"
 	"math/rand"
+	"os"
 	"time"
+
+	"github.com/dmacvicar/terraform-provider-libvirt/libvirt"
+	"github.com/hashicorp/terraform/plugin"
+	libvirtgo "github.com/libvirt/libvirt-go"
 )
 
+var version = "was not built correctly" // set via the Makefile
+
 func main() {
+	versionFlag := flag.Bool("version", false, "print version information and exit")
+	flag.Parse()
+	if *versionFlag {
+		err := printVersion()
+		if err != nil {
+			log.Fatal(err)
+		}
+		os.Exit(0)
+	}
+
 	defer libvirt.CleanupLibvirtConnections()
 
 	plugin.Serve(&plugin.ServeOpts{
 		ProviderFunc: libvirt.Provider,
 	})
 }
 
+func printVersion() error {
+	fmt.Printf("%s %s\n", os.Args[0], version)
+
+	fmt.Printf("Compiled against library: libvirt %s\n", parseVersion(libvirtgo.VERSION_NUMBER))
+
+	libvirtVersion, err := libvirtgo.GetVersion()
+	if err != nil {
+		return err
+	}
+	fmt.Printf("Using library: libvirt %s\n", parseVersion(libvirtVersion))
+
+	conn, err := libvirtgo.NewConnect("qemu:///system")
```

Comment:
here i dunno it might be tricky, since this variable is set via `main.tf` file. We can still say in the doc via ENV variable.

`URI=MY_AWESOME_REMOTE_KVM` terraform version 

( because otherwise we are giving always the local version)

Reply:
> We can still say in the doc via ENV variable.

Is there an existing variable I should use (like [AWS][1])?  Maybe something `virsh` also respects?  Also, should I use this new environment variable for both my new `-version` and the provider itself (like [AWS][1]), or should I only use it for `-version`? 

[1]: https://www.terraform.io/docs/providers/aws/#environment-variables

Reply:
for URI https://github.com/dmacvicar/terraform-provider-libvirt/blob/010069dd6b4b7c43e6a5fa40c22a8ccdb1828963/libvirt/provider.go#L17 we can use this one.

Reply:
>     DefaultFunc: schema.EnvDefaultFunc("LIBVIRT_DEFAULT_URI", nil),

Thanks for the link :).  Do you know why the default there is `nil` and not `qemu:///system`?

Reply:
Good question @dmacvicar  do you remember?

Reply:
I've pushed b8ff6d4 -> ef77653, which punts URI defaulting over to the C library.